sentry_telegram/plugin.py

```diff
@@ -145,9 +169,12 @@ def notify_users(self, group, event, fail_silently=False, **kwargs):
         self.logger.debug('Received notification for event: %s' % event)
         receivers = self.get_receivers(group.project)
         self.logger.debug('for receivers: %s' % ', '.join(receivers or ()))
+        receiver_threads = self.get_receiver_threads(group.project)
+        self.logger.debug('for receiver_threads: %s' % ', '.join(receiver_threads or ()))
         payload = self.build_message(group, event)
         self.logger.debug('Built payload: %s' % payload)
         url = self.build_url(group.project)
         self.logger.debug('Built url: %s' % url)
-        for receiver in receivers:
-            safe_execute(self.send_message, url, payload, receiver, _with_transaction=False)
+        for index, receiver in enumerate(receivers):
+            receiver_thread = receiver_threads[index]
```

Comment:
I see that there is a risk of an `IndexError` exception. In case a user enters fewer `receiver_threads` lines, than `receivers` lines. The built-in method `zip` might be helpful here.

Moreover, the current implementation makes it impossible to handle part of the receivers with a particular thread, while the remaining part - is without (for groups and direct messages, for example). Because in `get_receiver_threads()` all the empty lines will be filtered out. It will work for some simple cases with only one receiver and where all the receivers have/don't have a specific thread setting, but not for combinations of them.

I would suggest a slightly different approach, with setting `thread` at the same line as `receiver`'s id. So the receivers who need a particular thread would have it in their lines, while the remaining ones may stay unchanged.
Thread id may be separated by `/`, which seems to be a very intuitive way for "sub"-entities, and will be easy to work with from Python.
Example: `123456789` for a "direct" message without threading, and `123456789/098765` for a case with threading.

What would you say?

Reply:
Thanks for the feedback. yes, I completely agree with you. I also wanted to do it with an indication in the line with the group number, but I did not figure out how to implement it in python. This is my first python code. I develop professionally in ruby. Let's try this in our free time. Thank you